data/tutorials/language/0lg_09_env.md

```diff
@@ -0,0 +1,558 @@
+
+Since, by definition, effects lie outside function types, a function type can't reflect the effects a function may have. However, it is important to document the intended side effects a function may have. Consider the `Unix.time` function. It returns the number of seconds elapsed since Jan 1, 1970.
+```ocaml
+# Unix.time ;;
+- : unit -> float = <fun>
+```
+
+No data needs to be passed to this function in order to return its result. It is entirely context-dependent, i.e., entirely determined by external factors. Therefore, if it was passed something, it would be discarded. Note that it has to be passed something to trigger the computation. Requesting the information from the operating system, as suggested by the module name, is that computation. To sum up, since the function must receive data to trigger the computation but the data is going to be ignored, it makes sense to provide `()`, the unit value, the inhabitant of the type that only has a single value. What gets discarded is meaningless in the first place.
+
+A similar reasoning applies to functions producing an effect instead of being effect-determined functions. Consider `print_endline`. It prints the string it was passed to standard output, followed by a line termination.
+```ocaml
+# print_endline;;
+- : string -> unit = <fun>
+```
+Since the purpose of the function is only to produce an effect, it has no meaningful data to return; therefore, again, it makes sense to return the unit value.
+
+This illustrates the relationship between functions intended to have side effects and the unit type. The presence of the unit type does not indicate the presence of side effects. The absence of the unit type does not indicate the absence of side effects. But when no data needs to be passed as input or can be returned as output, the unit type should be used to indicate it and suggest the presence of side effects.
+
+## Conclusion
+
+Values, functions, and definitions are the most primitive and core concepts of OCaml. Even type-checking and execution aren't as central. The key concept is the environment. In summary, it is an ordered add-only key-value store. Add-only because nothing can be deleted. It is ordered because it is just the list of available definitions. When writing a `let`, zero, one, or several name-value pairs are added to the relevant environment. When calling a function parameter, names and arguments are also added to the environment. A closure is an environment-function pair.
```

Comment:
Maybe something like this: 

In essence, OCaml's fundamental concepts revolve around values, functions, and definitions, with type-checking and execution playing secondary roles. At the heart of OCaml lies the concept of the environment, a crucial element in its operation. To put it succinctly, the environment functions as an ordered, append-only key-value store, notable for its append-only nature, meaning that items cannot be removed. Furthermore, it maintains order by preserving the sequence of available definitions.

When we employ a `let `statement, we introduce zero, one, or more name-value pairs into the pertinent environment. Similarly, when invoking a function with its parameters, we extend the environment by adding names and their corresponding arguments, so a closure in OCaml embodies an environment-function pairing, cementing the interconnectedness of these fundamental concepts.